scanpullrequest/scanpullrequest.go

```diff
@@ -121,25 +120,31 @@ func scanPullRequest(repo *utils.Repository, client vcsclient.VcsClient) (err er
 	return
 }
 
-func toFailTaskStatus(repo *utils.Repository, issues *utils.IssuesCollection) bool {
+func toFailTaskStatus(repo *utils.Repository, issues *issues.ScansIssuesCollection) bool {
 	failFlagSet := repo.FailOnSecurityIssues != nil && *repo.FailOnSecurityIssues
-	return failFlagSet && issues.IssuesExists()
+	return failFlagSet && issues.IssuesExists(repo.PullRequestSecretComments)
 }
 
 // Downloads Pull Requests branches code and audits them
-func auditPullRequest(repoConfig *utils.Repository, client vcsclient.VcsClient) (issuesCollection *utils.IssuesCollection, err error) {
+func auditPullRequest(repoConfig *utils.Repository, client vcsclient.VcsClient) (issuesCollection *issues.ScansIssuesCollection, err error) {
+	repositoryInfo, err := client.GetRepositoryInfo(context.Background(), repoConfig.RepoOwner, repoConfig.RepoName)
```

Comment:
extract this logic to a function, while adding a check if the cloneUrl already exists in Repository.Params.Git.RepositoryCloneUrl.
Notice that GetRepositoryInfo might be called few times earlier in the flow (if we fetched config profile by url, if we are using GitHub and called verifyGitHubFrogbotEnvironment).
Please implement the following in all those places: whenever this function is called - please insert CloneInfo.HTTP to Repository.Params.Git.RepositoryCloneUrl. whenever we need to use this value - check there first and only if doesnt exists call GetRepositoryInfo (it can be in a case where we are not using github and bot using config profile with url).

Notice: in verifyGitHubFrogbotEnvironment we call GetRepositoryInfo for other perpouse but we can take advantage of this so set the cloneUrl.
Notice: the latest call to GetRepositoryInfo is in tryCheckoutToMostCommonAncestor. with your new call in earlier stage we will always have the cloneUrl at this point in Repository.Params.Git.RepositoryCloneUrl so we can delete the 'if' there and just take the value